docs/core/local-development.mdx

```diff
@@ -0,0 +1,349 @@
+await pool.query('INSERT INTO users (name) VALUES ($1)', ['Alice']);
+const result = await pool.query('SELECT * FROM users');
+console.log(result.rows);
+```
+
+## The `.suga` Directory
+
+Suga creates a `.suga` directory in your project for local state:
+
+```
+.suga/
+├── buckets/           # Bucket contents
+│   └── uploads/
+├── logs/              # Service logs
+│   ├── api.log
+│   └── worker.log
+└── dev.json          # Development server state
+```
+
+<Warning>
+  Add `.suga/` to your `.gitignore` - this directory contains local development state and should not be committed.
+</Warning>
+
+## Development Workflow
+
+### Starting Development
+
+```bash
+# Start all services and resources
+suga dev
+```
+
+### Making Code Changes
+
+1. **Edit your service code** - Changes are detected automatically
+2. **Service restarts** - Suga restarts the affected service
+3. **Test immediately** - New code is running
+
+### Debugging Services
+
+Standard debugging tools work normally:
+
+<Tabs>
+  <Tab title="Node.js">
+    ```json title="package.json"
+    {
+      "scripts": {
+        "dev": "node --inspect src/index.js"
+      }
+    }
+    ```
+
+    Then connect Chrome DevTools or VS Code debugger to `localhost:9229`.
+  </Tab>
+
+  <Tab title="Python">
+    ```yaml title="suga.yaml"
+    services:
+      api:
+        dev:
+          script: python -m debugpy --listen 5678 main.py
+    ```
+
+    Then connect VS Code or PyCharm debugger to `localhost:5678`.
+  </Tab>
+
+  <Tab title="Go">
+    ```bash title="Use delve debugger"
+    dlv debug --headless --listen=:2345 --api-version=2
+    ```
+
+    Then connect GoLand or VS Code debugger to `localhost:2345`.
+  </Tab>
+</Tabs>
+
+### Viewing Logs
+
+Service logs are written to `.suga/logs/`:
+
+```bash
+# Tail service logs
+tail -f .suga/logs/api.log
+
+# Or use your preferred log viewer
+cat .suga/logs/api.log | grep ERROR
+```
+
+### Seeding Local Data
+
+Pre-populate buckets for testing:
+
+```bash
+# Create test files
+mkdir -p .suga/buckets/uploads
+echo "Test content" > .suga/buckets/uploads/test.txt
+
+# Start dev server - files are available immediately
+suga dev
+```
+
+```typescript title="Read pre-seeded data"
+const data = await suga.uploads.read('test.txt');
+console.log(data); // "Test content"
+```
+
+## Environment Variables
+
+Services receive environment variables defined in `suga.yaml`:
+
+```yaml title="suga.yaml"
+services:
+  api:
+    env:
+      EXTERNAL_API_KEY: dev-key-12345
+      DEBUG: "true"
+    dev:
+      script: npm run dev
+
+databases:
+  main:
+    access:
+      api: [query]
+    env_var_key: DATABASE_URL  # Automatically injected during dev and production
+```
+
+During `suga dev`:
+- Service receives environment variables from `env` config
+- Database connection strings automatically injected based on `env_var_key`
+- Same variables are available in production (after Terraform deployment)
+- Use for configuration, API keys, feature flags
```

Comment:
Are we recommending that people use API keys and/or secrets in their `suga.yaml`?

```suggestion
- Use for configuration, feature flags, etc.
```